### PR TITLE
Python: Eliminate runtime package and use sqlalchemy

### PR DESCRIPTION
**Pull request by robholt**
Based on comments on my initial python codegen PR I've put together some potential changes to the generated python code.

I've marked this as a draft PR because I've only made manual edits to the authors example. Once we have a more solid idea of what the generated python code should look like I'll make the changes to the actual codegen code.

A quick overview of the changes:
- Use stdlib dataclasses instead of pydantic
- Use sqlalchemy as the execution engine
- Query functions are now inside of classes
- Eliminated the positional param issue, now `$1` is replaced with `:p1` instead of just `%s`.

Notes:
- Asyncio support in sqlalchemy requires the 1.4 release, which is currently in beta. Not really a long-term issue though, as it will eventually be out of beta. I've also confirmed the non-async code works fine with the latest stable 1.3.x release.
- The `:` character is special to sqlalchemy. The codegen will need to replace `:` with `\\:` in queries. I don't think this will cause issues, but there's always the potential for edge-case issues with regex replaces on the queries.

### CODE REVIEW

**Review by kyleconroy — CHANGES_REQUESTED** [review]
Really excited about this! Let me know if you have any questions about my comments.

File: examples/python/src/authors/query.py
Comment (robholt) [reply examples/python/src/authors/query.py:42]: Is that underlying problem actually a problem in python? You can't construct a dataclass (or call a query function) without providing values for all of the parameters that don't have explicit default values. If you translate the scenario in that issue to python you'd get both a type checker error and a runtime error with the current non-optional types.

File: examples/python/src/authors/query.py
Comment (robholt) [reply examples/python/src/authors/query.py:38]: I used the name of the sql file here, but I'm not sure that's the best option. Using `Querier` as a static name seems fine, although it means all the query files will have the same class names. That's not necessarily an issue as long as the module itself is imported instead of just the query class (so `from package import query` and `query.Querier()` instead of `from package.query import Querier` and `Querier()`).

File: examples/python/src/authors/query.py
Comment (robholt) [reply examples/python/src/authors/query.py:72]: :+1:

File: examples/python/src/authors/query.py
Comment (kyleconroy) [reply examples/python/src/authors/query.py:42]: That's a good a point about data classes. What if the class methods took keyword arguments instead of positional?

Comment (robholt) [reply examples/python/src/authors/query.py]: I like that idea, I've updated the methods to require keyword arguments

File: examples/python/src/authors/query.py
Comment (robholt) [reply examples/python/src/authors/query.py]: I ended up using the row indices instead of column names to support queries where some result columns have no name

File: examples/python/sqlc.json
Comment (kyleconroy) [inline examples/python/sqlc.json:13]: The "emit_*" style was something I didn't think much about when I first released sqlc. Feel free to pick different names if you'd like.

File: internal/config/config.go
Comment (kyleconroy) [inline internal/config/config.go:138]: Do overrides work yet? If not, let's just remove them for now.

**Review by kyleconroy — CHANGES_REQUESTED** [review]
Not sure why the tests are failing. Looks like you need to rebase as well. I'll hold off merging anything until this is green and ready to be merged in.

File: internal/config/config.go
Comment (robholt) [reply internal/config/config.go:138]: They do work! I haven't extensively tested it, but it's working great to override a few jsonb fields with a custom type in a different project.

File: examples/python/sqlc.json
Comment (robholt) [reply examples/python/sqlc.json:13]: The "emit_*" style seemed fine to me, nothing really better is coming to mind and it sticks with the style of other options.

**Comment by robholt** [comment]
Github actions was having issues yesterday, which caused the tests to fail. I've pushed the rebased commits which triggered a new test run.

**Review by kyleconroy — APPROVED** [review]

**Pull request merged**
